sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -436,13 +436,14 @@ cdef long count_points(Node* root, long count) nogil:
         # one point, and then the other neighboring cells
         # don't get filled in
     if DEBUGFLAG:
-        printf("[t-SNE] %i points in this node\n", count)
+        printf("[t-SNE] %li points in this node\n", count)
     return count
 
 
-cdef float compute_gradient(float[:,:] val_P,
+cdef float compute_gradient(float[:] val_P,
                             float[:,:] pos_reference,
-                            np.int64_t[:,:] neighbors,
+                            np.int64_t[:] neighbors,
+                            np.int64_t[:] indptr,
```

Comment:
If we really do have n_neighbors non-zero values in each row, I think the previous approach with a 2d array of (samples, neighbors) was better than having an indptr design. Why did you change it?